cmd/zoekt-sourcegraph-indexserver/sg.go

```diff
@@ -200,33 +201,40 @@ func (s *sourcegraphClient) List(ctx context.Context, indexed []uint32) (*Source
 
 		getIndexOptions := mkGetIndexOptionsFunc(tr)
 
-		// We ask the frontend to get index options in batches.
-		for repos := range batched(repos, batchSize) {
-			start := time.Now()
-			options, err := getIndexOptions(repos...)
-			duration := time.Since(start)
-
-			if err != nil {
-				metricResolveRevisionDuration.WithLabelValues("false").Observe(duration.Seconds())
-				tr.LazyPrintf("failed fetching options batch: %v", err)
-				tr.SetError()
+		// This does not scale well for large numbers of tenants with small numbers of
```

Comment:
Since we are already introducing the notion of "tenant ID" to the Sourcegraph `List` call, why not also introduce it to `GetIndexOptions`? That would provide a consistency in how we collect all the indexing information, instead of what we do now (only sometimes materializing tenant ID in the request/ response, but other times using `ctx`).